### PR TITLE
nixos/murmur: add option `environmentFile` for injecting secrets

### PR DESCRIPTION
**Pull request by elseym**
Secrets are injected from the environment into the rendered
configuration before each startup using envsubst.
The test now makes use of this feature for the server password.

###### Motivation for this change

Add the ability to pass secrets to the service without adding them to the Nix store.

analogous to #98550 and #97371

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by elseym** [comment]
@GrahamcOfBorg test mumble

File: nixos/modules/services/networking/murmur.nix [hunk 1]
```diff
@@ -256,14 +284,23 @@ in
...
-        User      = "murmur";
-        ExecStart = "${pkgs.murmur}/bin/murmurd -ini ${configFile}";
+        Type = if forking then "forking" else "simple";
+        PIDFile = mkIf forking "/run/murmur/murmurd.pid";
+        EnvironmentFile = mkIf (cfg.environmentFile != null) cfg.environmentFile;
+        ExecStart = "${pkgs.murmur}/bin/murmurd -ini /run/murmur/murmurd.ini";
+        Restart = "always";
+        RuntimeDirectory = "murmur";
+        RuntimeDirectoryMode = "0750";
+        SupplementaryGroups = [ "keys" ];
```
Comment (dasJ) [inline nixos/modules/services/networking/murmur.nix:302]: Umm, how is this related to the change?

Comment (elseym) [reply nixos/modules/services/networking/murmur.nix:302]: thanks for pointing this out, it's not related and shouldn't be there!

**Review by dasJ — APPROVED** [review]

**Review by lheckemann — APPROVED** [review]

File: nixos/modules/services/networking/murmur.nix [hunk 2]
```diff
@@ -256,14 +284,22 @@ in
...
-        RuntimeDirectory = mkIf forking "murmur";
-        User      = "murmur";
-        ExecStart = "${pkgs.murmur}/bin/murmurd -ini ${configFile}";
+        Type = if forking then "forking" else "simple";
+        PIDFile = mkIf forking "/run/murmur/murmurd.pid";
+        EnvironmentFile = mkIf (cfg.environmentFile != null) cfg.environmentFile;
+        ExecStart = "${pkgs.murmur}/bin/murmurd -ini /run/murmur/murmurd.ini";
+        Restart = "always";
+        RuntimeDirectory = "murmur";
+        RuntimeDirectoryMode = "0750";
```
Comment (lheckemann) [inline nixos/modules/services/networking/murmur.nix:301]: I don't think this is necessary?

Comment (elseym) [reply nixos/modules/services/networking/murmur.nix]: 👍
